datasafe-rest-impl/src/main/java/de/adorsys/datasafe/rest/impl/config/MultiDatasafeConfig.java

```diff
@@ -0,0 +1,120 @@
+package de.adorsys.datasafe.rest.impl.config;
+
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import de.adorsys.datasafe.business.impl.service.DaggerDefaultDatasafeServices;
+import de.adorsys.datasafe.business.impl.service.DaggerVersionedDatasafeServices;
+import de.adorsys.datasafe.business.impl.service.DefaultDatasafeServices;
+import de.adorsys.datasafe.business.impl.service.VersionedDatasafeServices;
+import de.adorsys.datasafe.directory.api.config.DFSConfig;
+import de.adorsys.datasafe.directory.impl.profile.config.MultiDFSConfig;
+import de.adorsys.datasafe.storage.api.SchemeDelegatingStorage;
+import de.adorsys.datasafe.storage.api.StorageService;
+import de.adorsys.datasafe.storage.impl.db.DatabaseConnectionRegistry;
+import de.adorsys.datasafe.storage.impl.db.DatabaseCredentials;
+import de.adorsys.datasafe.storage.impl.db.DatabaseStorageService;
+import de.adorsys.datasafe.storage.impl.fs.FileSystemStorageService;
+import de.adorsys.datasafe.storage.impl.s3.S3StorageService;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.context.annotation.Configuration;
+
+import javax.inject.Inject;
+import java.net.URI;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.Security;
+import java.util.Set;
+import java.util.concurrent.Executors;
+
+@Configuration
+@Conditional(MultiDfsCondition.class)
```

Comment:
It can be simplified to
@ConditionalOnProperty(name = "DATASAFE_SINGLE_STORAGE", havingValue="true")
and see comment above for single storage system config